### PR TITLE
Secondary Review

### PR DESCRIPTION
**Pull request by RathilVasani**
Primary Review is Complete , please proceed for Secondary Review

### CODE REVIEW

File: src/tests/suites/product/index.test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	afterEach((browser, done) => {
+		browser.execute("sessionStorage.clear()").pause(100);
+		done();
+	});
+
+	it("Apple And Samsung Filter", (browser) => {
+		browser.expect.elements(".shelf-item__title").count.to.equal(25);
+
+		browser
+			.click("xpath", "//span[contains(text(), 'Apple')]")
```
Comment (samiran-bs) [inline src/tests/suites/product/index.test.js:24]: input[value='Apple'] + span

File: src/tests/suites/product/index.test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+			.click("xpath", "//span[contains(text(), 'Samsung')]");
```
Comment (samiran-bs) [inline src/tests/suites/product/index.test.js:25]: input[value='Samsung'] + span

File: src/tests/suites/product/index.test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		// .pause(5000);
+
+		browser.expect.elements(".shelf-item").count.to.equal(16);
+	});
+
+	it("Apply 'Lowest to Highest' Order By Filter for prices", (browser) => {
+		browser
+			.click(".sort select option[value='lowestprice']")
+			.waitForElementPresent("xpath", "//*[@class = 'shelf-item'][1]")
```
Comment (samiran-bs) [inline src/tests/suites/product/index.test.js:34]: css: div.shelf-container-header + div.shelf-item

File: src/tests/suites/product/index.test.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+			.expect.element(
+				"/html/body/div/div/div/main/div[2]/div[2]/div[3]/div[1]/b",
```
Comment (samiran-bs) [inline src/tests/suites/product/index.test.js:36]: remove xpath.
possible css alternative
div.shelf-item__price > div.val > b
do a getText for this element.

File: src/tests/suites/product/index.test.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+		browser
+			.execute(
+				`browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"${
+					errors + failed - retries > 0 ? "failed" : "passed"
+				}","reason": "${errors} - errors ${failed} - failed - ${retries} - retried ${passed} - passed ${skipped} - skipped"}}`
+			)
+			.pause(1000)
+			.end();
+		done();
```
Comment (samiran-bs) [inline src/tests/suites/product/index.test.js:57]: repeated code. please add reusable function.

File: src/tests/suites/user/index.test.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+			.refresh()
+			.click("#signin")
+			.clearValue("#username input")
+			.setValue("#username input", "existing_orders_user")
+			.click(userData.existing_orders_user.selector)
+			.clearValue("#password input")
+			.setValue("#password input", userData.existing_orders_user.password)
+			.click(userData[userData.existing_orders_user.password].selector)
+			.click("#login-btn")
+			.pause(1000)
```
Comment (samiran-bs) [inline src/tests/suites/user/index.test.js:33]: can we perform page title or some other field assertion which does not exist on the prev page?

File: src/tests/suites/user/index.test.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
```
Comment (samiran-bs) [inline src/tests/suites/user/index.test.js:51]: can we perform page title or some other field assertion which does not exist on the prev page?

File: src/tests/suites/user/index.test.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+			.assert.containsText(".username", "existing_orders_user")
+			.click("xpath", "//p[text() = 'iPhone 12']/../div/button")
+			.click("#favourites")
+			.pause(1000)
```
Comment (samiran-bs) [inline src/tests/suites/user/index.test.js:55]: can we perform page title or some other field assertion which does not exist on the prev page?

File: src/tests/suites/user/index.test.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+			.clearValue("#username input")
+			.setValue("#username input", "existing_orders_user")
+			.click(userData.existing_orders_user.selector)
+			.clearValue("#password input")
+			.setValue("#password input", userData.existing_orders_user.password)
+			.click(userData[userData.existing_orders_user.password].selector)
+			.click("#login-btn")
+			.pause(1000)
+			.assert.containsText(".username", "existing_orders_user")
+			.click("xpath", "//p[text() = 'iPhone 12']/../div/button")
```
Comment (samiran-bs) [inline src/tests/suites/user/index.test.js:53]: change without xpath.

File: src/tests/suites/user/index.test.js (hunk 8 above)
Comment (samiran-bs) [inline src/tests/suites/user/index.test.js:55]: can we perform page title or some other field assertion which does not exist on the prev page?

File: src/tests/suites/user/index.test.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+
+		browser
+			.execute(
+				`browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"${
+					errors + failed - retries > 0 ? "failed" : "passed"
+				}","reason": "${errors} - errors ${failed} - failed - ${retries} - retried ${passed} - passed ${skipped} - skipped"}}`
+			)
+			.pause(1000)
+			.end();
+		done();
```
Comment (samiran-bs) [inline src/tests/suites/user/index.test.js:88]: reusable code.

File: yarn.lock [hunk 11]
```diff
@@ -0,0 +1,2709 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (samiran-bs) [inline yarn.lock:1]: is yarn.lock required to be checked-in ... isn't is auto generated?

Comment (NightWing1998) [reply yarn.lock:1]: yarn.lock is a peer dependency file like package-lock.json
We can omit this but there could be a good chance that in future some dependency change functionality and our tests will break

**Review by samiran-bs — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 12]
```diff
@@ -0,0 +1,49 @@
+version: "3"
```
Comment (samiran-bs) [inline docker-compose.yml:1]: update the docker-compose.yml file here using the Selenium 3.14 version which is working for TestNg.
and create a PR here to update this file too.

https://github.com/browserstack/browserstack-examples-reusable-artifacts/blob/main/templates/docker/selenium/docker-compose.yml

**Review by samiran-bs — CHANGES_REQUESTED** [review]

**Review by samiran-bs — APPROVED** [review]

**Pull request merged**